jbmc/regression/jbmc/deterministic_assignments_json/clinit-state.json

```diff
@@ -0,0 +1,424 @@
+      }
+    },
+    "concreteClassFirst":{
+      "@type":"MySubclassA"
+    },
+    "referenceToPrimitiveArray":{
+      "@id":2,
+      "@type":"[I",
+      "@items":[
+        5,55,610
+      ]
+    },
+    "interfaceSecond":{
+      "@type":"MyImplementationB",
+      "field":22
+    },
+    "primitiveArrayField":{"@ref":2},
+    "charArrayField":{
+      "@type":"[C",
+      "@items":[
+        "\u0000\u0001AB?;⿃"
+      ]
+    },
+    "fibCharField":{
+      "@type":"java.lang.Character",
+      "value":"ɢ"
+    },
+    "longField":{
+      "@type":"java.lang.Long",
+      "value":"202630"
+    },
+    "inheritedStatic":{
+      "@type":"java.lang.Integer",
+      "value":10
+    },
+    "interfaceContainerFirst":{
+      "@type":"MyInterfaceContainerA",
+      "field":{
+        "field":11
+      }
+    },
+    "genericWithInterfaceField":{
+      "@type":"GenericType",
+      "field":{
+        "@type":"MyImplementationA",
+        "field":11
+      }
+    },
+    "concreteClassSecond":{
+      "@type":"MySubclassB"
+    },
+    "genericMultiDimField":{
+      "@type":"GenericMultiDim",
+      "key":{
+        "@type":"Foo",
+        "i":2,
+        "b":{
+          "x":8
+        }
+      },
+      "value":{
+        "@type":"Bar",
+        "x":4
+      }
+    },
+    "specialCharField":{
+      "@type":"java.lang.Character",
+      "value":"\n"
+    },
+    "hexCharField":{
+      "@type":"java.lang.Character",
+      "value":"\u0001"
+    },
+    "shortField":{
+      "@type":"java.lang.Short",
+      "value":8
+    },
+    "genericInterfaceSimpleMore":{
+      "@type":"GenericImplementationSimpleMore",
+      "t":{
+        "@type":"Foo",
+        "i":2,
+        "b":{
+          "x":6
+        }
+      },
+      "u":{
+        "@type":"Bar",
+        "x":11
+      }
+    },
+    "interfaceContainerSecond":{
+      "@type":"MyInterfaceContainerB",
+      "field":{
+        "@type":"MyImplementationB",
+        "field":22
+      }
+    },
+    "referenceToReferenceArray":{"@ref":3},
+    "stringBuilderField":{
+      "@type":"java.lang.StringBuilder",
+      "value":"aaaaaaaa"
+    },
+    "myInheriting":{
+      "@type":"MyClassWithInheritedFields",
+      "childField":4,
+      "parentField":3,
+      "parentString":"parent"
+    },
+    "nullField":null,"abstractClassSecond":{
+      "@type":"MyAbstractImplementationB"
+    },
+    "referenceToOtherClass":{
+      "@id":1,
+      "@type":"Foo",
+      "i":12,
+      "b":{
+        "x":24
+      }
+    },
+    "stringField":{
+      "@type":"java.lang.String",
+      "value":"hello! hello! hello! hello! hello! hello! "
+    },
+    "referenceToString":{
+      "@type":"java.lang.String",
+      "value":"hello! hello! hello! hello! hello! hello! "
+    },
+    "enumField":{
+      "@type":"Color",
+      "name":"WHITE","ordinal":3,"grayscale":{
+        "@type":"java.lang.Boolean",
+        "value":true
+      }
+    }
+  },
+  "OtherClass":{
+    "otherField":{"@ref":1}
+  }
+}
```

Comment:
It would be great to include the json-io command line that generated that file. And also the json-io version. Maybe in a README in this folder.

Reply:
The version is mentioned in `assignments_from_json.h` but I can mention it again in the README. I did use a script to generate this file but it's messy and not really merge-able, and I did some manual changes after running it... I thought we decided that the script wasn't necessary for this PR? But if you think it would be useful, I'll tidy it up and add it to this commit.

Reply:
I think having the version here would be good. It's all about understanding where this file comes from, and how we can reproduce it. It's unfortunate that you had to do some manual tweaks. What is the extend of those?

Reply:
I added a script to generate something very close to `clinit-state.json` - minus some newlines, as the enum/String custom writers don't support pretty printing. I documented this in the README. The version of json-io is in the pom file of the newly added Maven project (for the script).